src/components/expanded-state/swap-details/SwapDetailsContractRow.js

```diff
@@ -0,0 +1,134 @@
+  copyAddress: 'copyAddress',
+  etherscan: 'etherscan',
+};
+
+const ContractActions = {
+  [ContractActionsEnum.copyAddress]: {
+    actionKey: ContractActionsEnum.copyAddress,
+    actionTitle: 'Copy Contract Address',
+    icon: {
+      iconType: 'SYSTEM',
+      iconValue: 'doc.on.doc',
+    },
+  },
+  [ContractActionsEnum.etherscan]: {
+    actionKey: ContractActionsEnum.etherscan,
+    actionTitle: 'View on Etherscan',
+    icon: {
+      iconType: 'SYSTEM',
+      iconValue: 'safari',
+    },
+  },
+};
+
+function useCopyContractAddress() {
+  const { setClipboard } = useClipboard();
+  const { onCopySwapDetailsText } = useContext(SwapDetailsContext);
+  return useCallback(
+    address => {
+      setClipboard(address);
+      onCopySwapDetailsText(address);
+    },
+    [onCopySwapDetailsText, setClipboard]
+  );
+}
+
+function SwapDetailsContractRowContent({
+  asset,
+  menuVisible,
+  scaleTo = 1.06,
+  ...props
+}) {
+  const colorForAsset = useColorForAsset(asset);
+  const animation = useTimingTransition(menuVisible, { duration: 150 });
+  const { addressColor, scale } = useMemoOne(
+    () => ({
+      addressColor: mixColor(animation, colors.blueGreyDark80, colorForAsset),
+      scale: interpolate(animation, {
+        inputRange: [0, 1],
+        outputRange: [1, scaleTo],
+      }),
+    }),
+    [animation, colorForAsset, scaleTo]
+  );
+
+  return (
+    <Animated.View style={{ transform: [{ scale }] }}>
+      <ButtonPressAnimation scaleTo={scaleTo} {...props}>
+        <SwapDetailsRow label={`${asset?.symbol} contract`}>
+          <SwapDetailsValue
+            address={asset?.address}
+            as={AnimatedTruncatedAddress}
+            color={addressColor}
+            firstSectionLength={6}
+          />
+          <SwapDetailsValue color={colors.blueGreyDark50}>
+            {` 􀁰`}
+          </SwapDetailsValue>
+        </SwapDetailsRow>
+      </ButtonPressAnimation>
+    </Animated.View>
+  );
+}
+
+export default function SwapDetailsContractRow({ asset, ...props }) {
+  const handleCopyContractAddress = useCopyContractAddress();
+
+  const menuConfig = useMemo(
+    () => ({
+      menuItems: [
+        ContractActions[ContractActionsEnum.etherscan],
+        {
+          ...ContractActions[ContractActionsEnum.copyAddress],
+          discoverabilityTitle: abbreviations.formatAddressForDisplay(
+            asset?.address
+          ),
+        },
+      ],
+      menuTitle: `${asset?.name} (${asset?.symbol}) token contract`,
+    }),
+    [asset]
+  );
+
+  const handlePressMenuItem = useCallback(
+    async ({ nativeEvent: { actionKey } }) => {
+      if (actionKey === ContractActionsEnum.copyAddress) {
+        handleCopyContractAddress(asset?.address);
+      } else if (actionKey === ContractActionsEnum.etherscan) {
+        await ethereumUtils.openTokenEtherscanURL(asset?.address);
+      }
+    },
+    [asset, handleCopyContractAddress]
+  );
+
+  return (
+    <ContextMenuButton
```

Comment:
what can we use here instead of on Android? Maybe it's something that can be unified with the helper menu on the transactions' list. Looks like a similar case.